### PR TITLE
Ccf test coverage

### PR DESCRIPTION
**Pull request by manushak**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

- add full test coverage
- separate test file to mock aws-use.json data
- make small code imporvments

### CODE REVIEW

**Review by narekhovhannisyan — APPROVED** [review]

File: src/lib/ccf/index.ts [hunk 1]
```diff
@@ -271,36 +271,27 @@ export const CloudCarbonFootprint = (
...
-      Skylake: () => {
-        architecture = 'Sky Lake';
-      },
-      Graviton: () => {
-        architecture = architecture.includes('2') ? 'Graviton2' : 'Graviton';
-      },
-      Unknown: () => {
-        architecture = 'Average';
-      },
+    let formattedArchitecture = architecture;
```
Comment (narekhovhannisyan) [inline src/lib/ccf/index.ts:274]:
```
const resolveAwsArchitecture = (architecture: string) => {
  const architectureFormat: {[key: string]: () => string} = {
    'AMD ': () => architecture.substring(4),
    Skylake: () => 'Sky Lake',
    Graviton: () => (architecture.includes('2') ? 'Graviton2' : 'Graviton'),
    Unknown: () => 'Average',
  };

  return Object.keys(architectureFormat).reduce((formattedArchitecture, key) => {
    if (architecture.includes(key)) {
      formattedArchitecture = architectureFormat[key]();
    }
    return formattedArchitecture;
  }, architecture);
};
```

Comment (narekhovhannisyan) [reply src/lib/ccf/index.ts:274]: for escaping using let

Comment (narekhovhannisyan) [reply src/lib/ccf/index.ts:274]:
```
const resolveAwsArchitecture = (architecture: string) => {
  const architectureFormat: {[key: string]: () => string} = {
    'AMD ': () => architecture.substring(4),
    Skylake: () => 'Sky Lake',
    Graviton: () => (architecture.includes('2') ? 'Graviton2' : 'Graviton'),
    Unknown: () => 'Average',
  };

  Object.keys(architectureFormat).forEach(key => {
    if (architecture.includes(key)) {
      architecture = architectureFormat[key]();
    }
  });

  return architecture;
};
```

Comment (narekhovhannisyan) [reply src/lib/ccf/index.ts:274]: this will work too

**Review by MariamKhalatova — APPROVED** [review]

**Pull request merged**